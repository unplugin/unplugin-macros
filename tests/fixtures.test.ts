import { resolve } from 'node:path'
import { rollupBuild, testFixtures } from '@sxzz/test-utils'
import { describe, vi } from 'vitest'
import Macros from '../src/rollup'

vi.spyOn(Math, 'random').mockReturnValue(0.5)

describe('fixture', async () => {
  await testFixtures(
    'tests/fixtures/*.{js,ts}',
    async (args, id) =>
      (
        await rollupBuild(id, [
          Macros(),
          {
            name: 'strip-types',
            transform(code) {
              return code.replaceAll('as any', '')
            },
          },
        ])
      ).snapshot,
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
    },
  )
})
