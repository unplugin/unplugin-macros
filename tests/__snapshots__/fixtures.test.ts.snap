// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixture > tests/fixtures/args.js 1`] = `
"// args.js
0 === 0;
"1" === '1';
false === false;
null === null;
undefined === undefined;
console.log({"a":"b"});
console.log([1,false,"hello"]);
"
`;

exports[`fixture > tests/fixtures/basic.js 1`] = `
"// basic.js
0.5 === 0.5;
1 === 1;
2 === 2;
3 === 3;
1 === 1;

const { bar = 1 } = {};
"
`;

exports[`fixture > tests/fixtures/css.js 1`] = `
"// css.js
// prettier-ignore
const style = {"color":"red"};
"
`;

exports[`fixture > tests/fixtures/ctx.js 1`] = `
"// ctx.js
"ctx.js" === 'ctx.js';
"
`;

exports[`fixture > tests/fixtures/dedent.js 1`] = `
"// dedent.js
const msg = "if (true) {\\n  const a = 'b'\\n}";
"
`;

exports[`fixture > tests/fixtures/import-namespace.js 1`] = `
"// import-namespace.js
0.5 === 0.5;
"
`;

exports[`fixture > tests/fixtures/nested-object.js 1`] = `
"// nested-object.js
"foo" === 'foo';
"
`;

exports[`fixture > tests/fixtures/node-builtin.js 1`] = `
"// node-builtin.js
"LE" === 'LE'
;("* text=auto eol=lf\\n") ===
  // prettier-ignore
  "* text=auto eol=lf\\n";
"
`;

exports[`fixture > tests/fixtures/non-macros.js 1`] = `
"// non-macros.js
{
  const getRandom = () => 2;
  getRandom() === 2;
}

''.toString();

const genHex = () =>
  Math.floor(Math.random() * 255)
    .toString(16)
    .padStart(2, '0');

const foo = {
  toString: '',
};

class TestCls {
  toString() {}
  #toString() {}
}
"
`;

exports[`fixture > tests/fixtures/promise.js 1`] = `
"// promise.js
const a = "ok";
const b = "ok";
"
`;

exports[`fixture > tests/fixtures/typescript.ts 1`] = `
"// typescript.js
10;
"
`;
